### ISSUE TITLE
autoComplete="cc-number" is not doing anything to allow autofill

### ISSUE THREAD

**Issue by Gambitboy** [facebook/react-native#36089]
### Description

I've trying to implement an TextInput for Credit Card Number that allows the users to autofill it using whatever password manager they use on their device.

I've tried different variations and props but its not working.

In my Snack it works when its running in web but not on Android or iOS.
I've also tested this on my own physical android device and it does not work.

Is this how its supposed to work or am I doing something wrong?

### Version

0.71.2

### Output of `npx react-native info`

System:
    OS: macOS 13.1
    CPU: (8) arm64 Apple M1 Pro
    Memory: 93.45 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.19.0 - ~/.nvm/versions/node/v16.19.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.19.0/bin/yarn
    npm: 8.19.3 - ~/.nvm/versions/node/v16.19.0/bin/npm
    Watchman: 2023.01.09.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: Not Found
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    Android SDK: Not Found
  IDEs:
    Android Studio: 2021.3 AI-213.7172.25.2113.9123335
    Xcode: 14.2/14C18 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.17 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.2.0 => 18.2.0
    react-native: 0.71.2 => 0.71.2
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

1. Init blank template
2. Create TextInput
3. Set autoComplete prop to "cc-number"

### Snack, code example, screenshot, or link to a repository

https://snack.expo.dev/x9JtUAYkj

**Comment by Gambitboy** [comment]
Bump, any assistance?

**Comment by Gambitboy** [comment]
I'm having the same issue with OTP autofill working in debug mode but doesnt work for release mode

**Comment by gunnartorfis** [comment]
Same here. I'm not sure what the expected behaviour is on iOS but on Android I'd assume it is meant to pretty well integrated. When using the prop with React Native Web, it gets auto-filled as expected on Chrome (Android).

**Comment by eminsr** [comment]
The issue persists in version 0.72.5 and `autoComplete="cc-number"`  also changes the `keyboardType`

**Comment by jhta** [comment]
same here

**Comment by simgill** [comment]
+

**Comment by phillipkey** [comment]
I'm seeing this issue on React Native 0.69.0. The fields work fine for android but nothing on iOS.

 I've been searching to see if there's something that needs to be added to the app's configuration to enable the permission since I believe it's using the keychain.

**Comment by Gambitboy** [comment]
Android is working on 0.76.1

**Comment by Gambitboy** [comment]
iOS still broken

**Comment by euroclydon37** [comment]
I've been trying to research this, and it seems to me like credit card autofill is only supported in Safari. I built a barebones app in Xcode to test it, and setting the `textContentType` to `.fullStreetAddress` makes it suggest my address, but setting it to `.creditCardNumber` does nothing except change the keyboard type. Here's the code for root view of the app.
```swift
import SwiftUI

struct ContentView: View {
    @State private var text = ""
    
    var body: some View {
        VStack {
            TextField("Card number", text: self.$text).textContentType(.creditCardNumber)
        }
        .padding()
    }
}
```

Does anyone know of an app they've used that _did_ support autofill? I would love to be proven wrong.

**Comment by chohra-med** [comment]
Still same issues in RN 0.74

**Comment by euroclydon37** [comment]
I don't think this is a React Native issue, per my comment above. It doesn't even work in Swift. Until someone presents an example of an iOS app successfully implementing cc-autocomplete, I'm inclined to believe it's either a bug on Apple's end or they're purposefully restricting cc autocomplete to Safari.

**Comment by AamirMir99** [comment]
I agree with @euroclydon37 Apple is intentionally restricting credit card autocomplete to Safari and WebViews. I’ve tried every possible approach, but autofill never includes credit card details in native apps. Over the years, Apple has significantly improved the convenience of Apple Pay, but it now feels like they’re making it increasingly difficult to use alternative payment methods.

**Comment by Bezaatay** [comment]
same
